web/src/reducers/patch.js

```diff
@@ -0,0 +1,33 @@
+import u from 'updeep';
+
+import { SAVE_APD_SUCCESS } from '../actions/apd';
+import { EDIT_APD } from '../actions/editApd';
+
+const initialState = [];
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case SAVE_APD_SUCCESS:
+      return initialState;
+
+    case EDIT_APD: {
+      const existingIndex = state.findIndex(
+        ({ op, path }) => op === 'replace' && path === action.path
+      );
+      if (existingIndex < 0) {
```

Comment:
I should put comments in the code about this, but the gist is if there's already a patch to update this specific path, we don't need another one, we can just update the existing one.

But now that I think about it, I don't think this is a fully correct implementation, because if I update the 5th element in an array, delete the fifth element in an array, and then update the new 5th element in the array, that is different than updating the 5th element twice and then deleting it. So, probably need to go back and rethink this. Maybe update the existing patch if no add or remove operations occur later in the patch set. 🤔